src/libraries/System.Private.CoreLib/src/System/Number.DecimalIeee754.cs

```diff
@@ -0,0 +1,352 @@
+                    {
+                        return -1;
+                    }
+                    if (remainder > TValue.Zero)
+                    {
+                        return -1;
+                    }
+                    return 0;
+                }
+
+                return 1;
+            }
+        }
+
+        private static unsafe TValue NumberToDecimalIeee754Bits<TDecimal, TValue>(ref NumberBuffer number)
+            where TDecimal : unmanaged, IDecimalIeee754ParseAndFormatInfo<TDecimal, TValue>
+            where TValue : unmanaged, IBinaryInteger<TValue>
+        {
+            Debug.Assert(number.DigitsPtr[0] != '0');
+            Debug.Assert(number.DigitsCount != 0);
+
+            TValue significand;
+            int exponent;
+
+            if (number.DigitsCount > TDecimal.Precision)
+            {
+                DecimalIeee754Rounding<TDecimal, TValue>(ref number, TDecimal.Precision, out significand, out exponent);
+
+                if (exponent > TDecimal.MaxExponent)
+                {
+                    return number.IsNegative ? TDecimal.NegativeInfinity : TDecimal.PositiveInfinity;
+                }
+            }
+            else
+            {
+                int positiveExponent = (Math.Max(0, number.Scale));
+                int integerDigitsPresent = Math.Min(positiveExponent, number.DigitsCount);
+                int fractionalDigitsPresent = number.DigitsCount - integerDigitsPresent;
+                exponent = number.Scale - integerDigitsPresent - fractionalDigitsPresent;
+
+                if (exponent < TDecimal.MinExponent)
+                {
+                    return SmallExponentRounding(ref number, exponent);
+                }
+
+                significand = TDecimal.NumberToSignificand(ref number, number.DigitsCount);
+                return DecimalIeee754BinaryEncoding<TDecimal, TValue>(number.IsNegative, significand, exponent);
+            }
+            return DecimalIeee754BinaryEncoding<TDecimal, TValue>(number.IsNegative, significand, exponent);
+
+            static TValue SmallExponentRounding(ref NumberBuffer number, int exponent)
+            {
+                Debug.Assert(exponent < TDecimal.MinExponent);
+                int numberDigitsRemove = (TDecimal.MinExponent - exponent);
+                if (numberDigitsRemove < number.DigitsCount)
+                {
+                    int numberDigitsRemain = number.DigitsCount - numberDigitsRemove;
+                    DecimalIeee754Rounding<TDecimal, TValue>(ref number, numberDigitsRemain, out TValue significand, out exponent);
+                    return DecimalIeee754BinaryEncoding<TDecimal, TValue>(number.IsNegative, significand, exponent);
+                }
+                else
+                {
+                    return TDecimal.Zero;
+                }
+            }
+        }
+
+        private static unsafe TValue DecimalIeee754BinaryEncoding<TDecimal, TValue>(bool signed, TValue significand, int exponent)
```

Comment:
nit: exponent and significand are swapped compared to the typical order of such APIs

Reply:
These types of APIs, even though they are private/internal, likely also benefit from XML doc comments to be explicit that that `significand` is the "full" (non-trailing) significand and `exponent` is the unbiased exponent.

The terms get interchanged a lot and so being explicit can help ensure future readers understand the intent.